rosidl_typesupport_opendds_cpp/rosidl_typesupport_opendds_cpp/__init__.py

```diff
@@ -15,9 +15,12 @@
 import os
 import subprocess
 import sys
+import shutil
 
 from rosidl_cmake import generate_files
-
+from pathlib import Path
```

Comment:
It doesn't look like `pathlib.Path` isn't being used. I do highly recommend using it for path abstraction, because it is very useful.

Reply:
I had added that for some calls I realized later I didn't need to make and forgot to remove. Most of code here is from the original RTI file, that does not use pathlib.Path. Are there specific calls that are better made with it than with os.path?

Reply:
In my opinion `pathlib` is a better design. [It combines many string based functions from the `os` module](https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module) into single object oriented interface. They are more or less similar in functionality however, so isn't necessarily wrong to go with the older string based functions. It is nice to be able to combine paths using `/` though:

```python3
>>> from pathlib import Path
>>> x = Path('a')
>>> x / 'b'
PosixPath('a/b')
```

Reply:
Isn't pathlib also OS agnostic? Currently we're building this all in a container but ultimately the build will be at least for Windows, Linux, and Mac.

Reply:
Both have separate Windows and Unix implementations, so I don't think they don't have any advantage over on another in that regard.

Reply:
I was thinking more about `/` path delimiter vs `\`.